oak/server/rust/oak_runtime/src/metrics/server.rs

```diff
@@ -53,25 +56,43 @@ async fn serve_metrics(_req: Request<Body>) -> Result<Response<Body>, MetricsSer
     info!("Metrics size: {}", buffer.len());
 
     Response::builder()
-        .status(http::StatusCode::OK)
+        .status(StatusCode::OK)
         .header(CONTENT_TYPE, encoder.format_type())
         .body(Body::from(buffer))
         .map_err(|e| {
             MetricsServerError::ResponseError(format!("Could not build the response: {}", e))
         })
 }
 
-async fn make_server(port: u16) -> Result<(), hyper::error::Error> {
+async fn serve_metrics(
+    runtime: Arc<Runtime>,
+    req: Request<Body>,
+) -> Result<Response<Body>, MetricsServerError> {
+    match (req.method(), req.uri().path()) {
+        (&Method::GET, "/metrics") => handle_metrics_request(runtime).await,
+        _ => Ok(Response::builder()
+            .status(StatusCode::NOT_FOUND)
+            .body(Body::from("Not Found!\n"))
+            .unwrap()),
+    }
+}
+
+async fn make_server(runtime: Arc<Runtime>, port: u16) -> Result<(), hyper::error::Error> {
     let addr = SocketAddr::from(([0, 0, 0, 0], port));
 
     // A `Service` is needed for every connection, so this
     // creates one from the `process_metrics` function.
-    let make_svc = make_service_fn(|_conn| async {
-        // service_fn converts our function into a `Service`
-        Ok::<_, hyper::Error>(service_fn(serve_metrics))
+    let make_service = make_service_fn(move |_| {
```

Comment:
I think it's useful to keep the parameter name as `_conn`, for documentation purposes.